### PR TITLE
ISSUE-539 Apply fuzzy search

### PR DESCRIPTION
**Pull request by hungphan227**
## Summary by CodeRabbit

* **New Features**
  * Added a fuzzy search option for calendar event searches (new configuration key, disabled by default).

* **Tests**
  * Added parameterized tests to verify fuzzy search behavior and edge cases.

* **Chores**
  * Added a test-scoped dependency to support updated tests.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: storage-opensearch/src/main/java/com/linagora/calendar/storage/opensearch/OpensearchCalendarSearchService.java
Comment (chibenwa) [inline storage-opensearch/src/main/java/com/linagora/calendar/storage/opensearch/OpensearchCalendarSearchService.java]:
```suggestion
    private Optional<Query> buildQuery(EventSearchQuery searchRequest) {
```

I do not understand why we have "Keyword" in there

File: storage-opensearch/src/test/java/com/linagora/calendar/storage/opensearch/OpensearchCalendarSearchServiceTest.java [hunk 1]
```diff
@@ -102,4 +106,28 @@ void searchShouldOnlyMatchSummaryPrefixThatIsNotOverMaxNgramLength() {
             assertThat(searchResults2).hasSize(0);
         });
     }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"sprant", "plenning", "print", "splanning", "sphant"})
+    void searchShouldSupportFuzziness(String search) {
```
[the remark this answers is not in the document]
Comment (hungphan227) [reply storage-opensearch/src/test/java/com/linagora/calendar/storage/opensearch/OpensearchCalendarSearchServiceTest.java]: updated
